### DISCUSSION TITLE
Integrate MapView to MapControl

### DISCUSSION THREAD

**Discussion by charlenni** [Mapsui/Mapsui#2391] [Development]
I started with a first step in this direction, but I'm not sure, what users expect from such an integration. What is the advantage of MapView against MapControl? Coordinate system? MapControl with preinstalled widgets? Easy way to add markers/pins? Or drawables like lines and rectangles?

**Comment by pauldendulk** [comment]
I think working with with Map extensions will make this very flexible. We can have small grained and course grained methods. The course grained methods could generate the entire MapView will one call, but since the current MapView contains a lot, so perhaps users not to combine a few methods to compose the entire MapView.

The most important thing I want to say is that this is not something we have to worry about right now. We need to get proper lego blocks, we need a way to compose lego blocks. Exactly how we compose things is the last step. It should be easy to create on version or another.

**Comment by charlenni** [comment]
Ok, then we have the same line here. I will work on the Markers and, if @janusw added the GroupingWidget, I will work on the MapExtensions for the MapView layout.

**Reply by pauldendulk** [reply]
Yes, but 'working' in this phase mostly means thinking about the design. Like you do here https://github.com/Mapsui/Mapsui/issues/2387#issuecomment-1892574096. Although hands on work is also needed to understand things. When starting on the MapControl touch logic I first did a number of innocent cleanup PRs get in the touch with the code

https://github.com/Mapsui/Mapsui/pull/2382
https://github.com/Mapsui/Mapsui/pull/2383
https://github.com/Mapsui/Mapsui/pull/2388

**Reply by charlenni** [reply]
Yes, but thinking alone wouldn't bring us further.

I started this topic to get some information, why people use MapView instead of MapControl. But your answer was the only one. And as I said earlier, MapView is for me only a convinience layer over MapControl. You don't have to understand what features and styles are for and you have a nice looking map with the default buttons you expect.

The basics are discussed: all this works by map extensions. No changes to MapControl are made (except the few we make on the way ;-) You could use these extensions or not, the map could be made by hand too. 

What I introduced so far: 
- MapExtensions to create a map in a fluent way. 
- New class for a Marker, a combination of PointFeature (from where it derives) and two different styles (image and callout)

That's it.
